accelerator/x86_accelerator.py

```diff
@@ -0,0 +1,298 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import os
+import pkgutil
+import importlib
+
+from .abstract_accelerator import DeepSpeedAccelerator
+# During setup stage torch may not be installed, pass on no torch will
+# allow op builder related API to be executed.
+try:
+    import torch.cpu
+except ImportError:
+    pass
+
+
+class x86_Accelerator(DeepSpeedAccelerator):
+
+    def __init__(self):
+        self._name = 'x86'
+
+    def is_synchronized_device(self):
+        return False
+
+    # Device APIs
+    def device_name(self, device_index=None):
+        if device_index is None:
+            return 'x86'
+        return 'x86:{}'.format(device_index)
+
+    def device(self, device_index=None):
+        return torch.device(device_index)
+
+    def set_device(self, device_index):
+        torch.cpu.set_device(device_index)
+
+    def current_device(self):
+        return torch.cpu.current_device()
+
+    def current_device_name(self):
+        return 'x86:{}'.format(torch.cpu.current_device())
+
+    def device_count(self):
+        return torch.cpu.device_count()
+
+    def synchronize(self, device_index=None):
+        return torch.cpu.synchronize(device_index)
+
+    # RNG APIs
+    def random(self):
+        return torch.random
+
+    def set_rng_state(self, new_state, device_index=None):
+        if device_index is None:
+            return torch.set_rng_state(new_state)
+
+        return torch.set_rng_state(new_state, device_index)
+
+    def get_rng_state(self, device_index=None):
+        if device_index is None:
+            return torch.get_rng_state()
+
+        return torch.get_rng_state(device_index)
+
+    def manual_seed(self, seed):
+        return torch.manual_seed(seed)
+
+    def manual_seed_all(self, seed):
+        return torch.manual_seed_all(seed)
+
+    def initial_seed(self, seed):
+        return torch.initial_seed(seed)
+
+    def default_generator(self, device_index):
+        return torch.default_generators[device_index]
+
+    # Streams/Events
+    @property
+    def Stream(self):
+        return torch.cpu.Stream
+
+    def stream(self, stream):
+        return torch.cpu.stream(stream)
+
+    def current_stream(self, device_index=None):
+        return None
+
+    def default_stream(self, device_index=None):
+        return None
+
+    @property
+    def Event(self):
+        return None
+
+    # Memory management
+    def empty_cache(self):
+        return
+
+    def memory_allocated(self, device_index=None):
```

Comment:
@mrwyattii - do we want to do this with psutil the same way the current cpu accelerator does?

Reply:
Yes, I think that is the right thing. I don't think torch has any cpu-specific memory information available like for cuda devices

Reply:
Sounds good, updated.